### ISSUE TITLE
discrete fixation map

### ISSUE THREAD

**Issue by DilaraAlbayrak** [CZHQuality/Sal-CFS-GAN#5]
I have only fixation density maps and rgb images which stands for dense saliency map and image of the required input. Since I dont have discrete fixation map, how can I train without discrete fixation map?

**Comment by CZHQuality** [comment]
Hi DilaraAlbayrak,

Thanks for your attention.

In this repository, the discrete fixation maps are exploited to compute the NSS loss. 
Notably, this repository supports multiple loss functions, such as CC, KLD, and L_1 losses, which only need the dense gaze maps. Therefore, you are able to train the model without using NSS loss, but the performance might be unsatisfying. 

There are two solutions to abandon the NSS loss:
1. You can add a simple command after the default training command, as follows: 
"python3 train.py --name label2city_512p --no_instance --label_nc 0 --no_ganFeat_loss --netG global_unet --resize_or_crop none --no_nss_loss"   (recommended) 
2. You can visit the source code at "\options\train_options.py", and reset the weights w.r.t. NSS loss as zero,  i.e. "self.parser.add_argument('--lambda_NSS', type=float, default=1.0, help='weight for NSS loss')" --> "self.parser.add_argument('--lambda_NSS', type=float, default=0.0, help='weight for NSS loss')". Then, please replace the input w.r.t. the discrete map by the dense fixation map, thus is has no impact on final result while causing no bugs  (not recommended) 

If you are using some popular open-source datasets to train your model from scratch, such as MIT1003, CAT2000, SALICON, the discrete fixation maps are available here:

MIT1003: 
http://people.csail.mit.edu/tjudd/WherePeopleLook/index.html

CAT2000:
http://saliency.mit.edu/trainSet.zip

SALICON:
http://salicon.net/challenge-2017/

If you have any question, please don't hesitate to contact with me.

**Comment by DilaraAlbayrak** [comment]
Thanks for your answer, I'll try without NSS loss.

I don't use these datasets, thanks for redirecting, though. I also would like to understand the discrete fixation map term, in the first page (MIT1003) there is a figure which I have copied below, is the discrete fixation map the last column that I enclosed yellow? It seems more like continuous, but I don't know what discrete fixation map is, white dots stand for eye fixation points?

![image](https://user-images.githubusercontent.com/7215154/73476894-3ae4eb80-43a4-11ea-883d-593ce8ab21db.png)

**Comment by CZHQuality** [comment]
Hi DilaraAlbayrak,

The figure that you mentioned above (highlighnted by yellow circle) is NOT the discrete fixation map. It seems like a binary annotation for semantic segmentation task (the salient foreground versus the non-salient background) .

The discrete fixation map, actually, is the vanilla/raw format of the human eye-movement data captured by the eye-tracker. The discrete fixation map only captures a very small fraction of the discrete pixels that represent the human gaze points when viewing this image. For example, if an image contains 400X600 pixels, the discrete fixation map only contains around 80 (not very rigid, around 60-150 pixels) non-zero pixels. 

For most of the open-source datasets, the discrete human fixation maps are presented by ".mat" format, where you can observe the values of each pixels directly. Notice that the values of most of the background pixels are zero. You can also transfer the format from ".mat" to ".jpg"/".png"/".bmp" for better observation and better implementation in Pytorch ...

For better explanation, I provide some simple demo-samples for your reference, as follows:
[demos.zip](https://github.com/CZHQuality/Sal-CFS-GAN/files/4137775/demos.zip)

Notice that the dense fixation maps are obtained by the Gaussian blur operation over the raw discrete fixation map.

**Comment by DilaraAlbayrak** [comment]
I am very grateful for your informative answers and examples. Then, I see that discrete fixation maps contains only zero or one (white) pixels, there is no gray-scale values between zero and one. Fixated pixels are assigned to one whereas other pixels are assigned to zero.
